pages/index.tsx

```diff
@@ -23,18 +26,34 @@ export default function Index(props: {
     return <WelcomeScreen handleSignup={handleSignup} />
   }
 
-  if (!adminUser && (!apiUser || !apiUser.hasConsentedToTerms)) {
+  if (
+    API_MANAGER_ENABLED &&
+    !adminUser &&
+    (!apiUser || !apiUser.hasConsentedToTerms)
+  ) {
     // New API user sign up will have both adminUser and apiUser to null,
     // or apiUser will have the terms not accepted.
     // For these users, display the API setup component.
     return <ApiUserForm createApiUser={createApiUser} />
   }
   if (adminUser) return <AdminUserDashboard />
+
+  if (API_MANAGER_ENABLED && apiUser)
+    return (
+      <ApiUserDashboard
+        apiUser={apiUser}
+        clearWelcome={() => push('/')}
+        showWelcome={query && !!query.newApiAccount} // If an API user has just been created, show welcome message.
+      />
+    )
+
   return (
-    <ApiUserDashboard
-      apiUser={apiUser}
-      clearWelcome={() => push('/')}
-      showWelcome={query && !!query.newApiAccount} // If an API user has just been created, show welcome message.
-    />
+    <>
+      <b>User is of invalid type!</b>
+      <p>
+        This is likely because the API module is disabled although the user is
```

Comment:
It doesn't have to be the API module, is it?

Reply:
It does not you're correct. Reworded in 3a29c33ffe0b1276cc11e44d7ce6ffe766f99d71